dist/tools/buildsystem_sanity_check/check.sh

```diff
@@ -362,6 +362,31 @@ check_stderr_null() {
         | error_with_message "Redirecting stderr and stdout to /dev/null is \`>/dev/null 2>&1\`; the other way round puts the old stderr to the new stdout."
 }
 
+# Test application directories that starts by the following strings are invalid
+# and should moved to the corresponding tests subdirectory.
+FORBIDDEN_TEST_DIRS=()
+FORBIDDEN_TEST_DIRS+=('build_system')
+FORBIDDEN_TEST_DIRS+=('core')       # => move under tests/core
+FORBIDDEN_TEST_DIRS+=('cpu')        # => move under tests/cpu
+FORBIDDEN_TEST_DIRS+=('driver')     # => move under tests/drivers
+FORBIDDEN_TEST_DIRS+=('gnrc')       # => move under tests/net
+FORBIDDEN_TEST_DIRS+=('periph')     # => move under tests/periph
+FORBIDDEN_TEST_DIRS+=('net')        # => move under tests/net
+FORBIDDEN_TEST_DIRS+=('pkg')        # => move under tests/pkg
+FORBIDDEN_TEST_DIRS+=('vfs')        # => move under tests/sys
+FORBIDDEN_TEST_DIRS+=('xtimer')     # => move under tests/sys
+FORBIDDEN_TEST_DIRS+=('ztimer')     # => move under tests/sys
+
+check_tests_application_path() {
+    local patterns=()
+    patterns+=(-regex "^tests/bench_.*/Makefile")
+    for forbidden in "${FORBIDDEN_TEST_DIRS[@]}"; do
+        patterns+=(-o -regex "^tests/${forbidden}_.*/Makefile")
+    done
+
+    find tests/ -type f ${patterns[@]} | error_with_message "Invalid application path in tests/"
```

Comment:
```suggestion
    find tests/ -type f "${patterns[@]}" | error_with_message "Invalid application path in tests/"
```

If I recall correctly, that would quote the array elements, not the whole array.

Reply:
Tested locally and you are right. I also amended a forbidden path for `tests/sys_`.